### ISSUE TITLE
Address already in use WARNING under linux

### ISSUE THREAD

**Issue by kashif** [pgbouncer/pgbouncer#2]
Hi,

I get the following error when I set the: `listen_addr = *` option and compile with `--disable-evdns` flag:

```
$ sudo -u pgbouncer ./pgbouncer -v -v -v /etc/pgbouncer.conf
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: [databases]
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: '*' = 'host=/var/run/postgresql port=5432 pool_size=64'
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: '*' = 'host=/var/run/postgresql port=5432 pool_size=64' ok:1
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: [pgbouncer]
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: 'logfile' = '/var/log/pgbouncer/pgbouncer.log'
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: 'logfile' = '/var/log/pgbouncer/pgbouncer.log' ok:1
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: 'pidfile' = '/var/run/pgbouncer/pgbouncer.pid'
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: 'pidfile' = '/var/run/pgbouncer/pgbouncer.pid' ok:1
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: 'listen_addr' = '*'
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: 'listen_addr' = '*' ok:1
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: 'listen_port' = '6432'
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: 'listen_port' = '6432' ok:1
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: 'unix_socket_dir' = '/var/run/pgbouncer'
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: 'unix_socket_dir' = '/var/run/pgbouncer' ok:1
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: 'auth_type' = 'md5'
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: 'auth_type' = 'md5' ok:1
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: 'auth_file' = '/etc/userlist.txt'
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: 'auth_file' = '/etc/userlist.txt' ok:1
2011-05-09 16:41:28.870 25986 DEBUG parse_ini_file: 'admin_users' = 'postgres'
2011-05-09 16:41:28.871 25986 DEBUG parse_ini_file: 'admin_users' = 'postgres' ok:1
2011-05-09 16:41:28.871 25986 DEBUG parse_ini_file: 'stats_users' = 'postgres'
2011-05-09 16:41:28.871 25986 DEBUG parse_ini_file: 'stats_users' = 'postgres' ok:1
2011-05-09 16:41:28.871 25986 DEBUG parse_ini_file: 'pool_mode' = 'session'
2011-05-09 16:41:28.871 25986 DEBUG parse_ini_file: 'pool_mode' = 'session' ok:1
2011-05-09 16:41:28.871 25986 DEBUG parse_ini_file: 'server_reset_query' = 'DISCARD ALL'
2011-05-09 16:41:28.871 25986 DEBUG parse_ini_file: 'server_reset_query' = 'DISCARD ALL' ok:1
2011-05-09 16:41:28.871 25986 DEBUG parse_ini_file: 'ignore_startup_parameters' = 'extra_float_digits'
2011-05-09 16:41:28.871 25986 DEBUG parse_ini_file: 'ignore_startup_parameters' = 'extra_float_digits' ok:1
2011-05-09 16:41:28.871 25986 DEBUG parse_ini_file: 'max_client_conn' = '100'
2011-05-09 16:41:28.871 25986 DEBUG parse_ini_file: 'max_client_conn' = '100' ok:1
2011-05-09 16:41:28.871 25986 DEBUG parse_ini_file: 'default_pool_size' = '20'
2011-05-09 16:41:28.871 25986 DEBUG parse_ini_file: 'default_pool_size' = '20' ok:1
2011-05-09 16:41:28.871 25986 NOISE event: 100, SBuf: 124, PgSocket: 260, IOBuf: 2060
2011-05-09 16:41:28.871 25986 LOG File descriptor limit: 1024 (H:1024), max_client_conn: 100, max fds possible: 110
2011-05-09 16:41:28.871 25986 DEBUG pktbuf_dynamic(128): 0x806a5b0
2011-05-09 16:41:28.871 25986 DEBUG make_room(0x806a5b0, 9): realloc newlen=256
2011-05-09 16:41:28.871 25986 DEBUG pktbuf_dynamic(128): 0x80693a0
2011-05-09 16:41:28.872 25986 NOISE connect(4, unix:/var/run/pgbouncer/.s.PGSQL.6432) = Connection refused
2011-05-09 16:41:28.872 25986 LOG Stale pidfile, removing
2011-05-09 16:41:28.873 25986 DEBUG add_listen: 0.0.0.0:6432
2011-05-09 16:41:28.873 25986 NOISE old TCP_DEFER_ACCEPT on 8 = 0
2011-05-09 16:41:28.873 25986 NOISE install TCP_DEFER_ACCEPT on 8
2011-05-09 16:41:28.873 25986 LOG listening on 0.0.0.0:6432
2011-05-09 16:41:28.873 25986 DEBUG add_listen: ::/6432
2011-05-09 16:41:28.873 25986 WARNING Cannot listen on ::/6432: bind(): Address already in use
2011-05-09 16:41:28.873 25986 DEBUG add_listen: unix:/var/run/pgbouncer/.s.PGSQL.6432
2011-05-09 16:41:28.873 25986 LOG listening on unix:/var/run/pgbouncer/.s.PGSQL.6432
2011-05-09 16:41:28.873 25986 LOG process up: pgbouncer 1.4.1, libevent 2.0.11-stable (epoll), adns: libc-2.11
```

I get the same error when using `--enable-evdns` as well. I am using `libevent-2.0.11` under Linux 32-bit version 2.6.35.

**Comment by DieMyst** [comment]
i have same problem with Linux 3.10.0-123.el7.x86_64 #1 SMP Mon May 5 11:16:57 EDT 2014 x86_64 x86_64 x86_64 GNU/Linux
found old issue like that http://t1510.db-postgresql-pgbouncer-general.dbtalk.info/issue-when-listen-addr-is-set-to-t1510.html , but can't see attach of patch in answer.
Try to change ipv6_only value in linux, but nothing changes. Will this issue be considered?

**Comment by markokr** [comment]
It should be fixed in 1.6rc1
